generator/generator.js

```diff
@@ -18,53 +24,159 @@ program
     new Option("-f, --force", "force creation (might overwrite existing files)")
   )
   .addOption(new Option("-cfg, --configEngine", "extends config engine"))
+  .addOption(new Option("-ch, --chains <letters...>").choices(AVAILABLE_CHAINS))
   .addOption(
-    new Option(
-      "-name, --name <string>",
-      "name of the proposal (e.g. CapsIncrease)"
-    ).makeOptionMandatory()
-  )
-  .addOption(
-    new Option("-ch, --chains <letters...>")
-      .choices([
-        "Ethereum",
-        "Optimism",
-        "Arbitrum",
-        "Polygon",
-        "Avalanche",
-        "Fantom",
-        "Harmony",
-        "Metis",
-      ])
-      .makeOptionMandatory()
-  )
-  .addOption(
-    new Option("-pv, --protocolVersion <string>")
-      .choices(["V2", "V3"])
-      .makeOptionMandatory()
+    new Option("-pv, --protocolVersion <string>").choices(AVAILABLE_VERSIONS)
   )
   .addOption(new Option("-t, --title <string>", "aip title"))
   .addOption(new Option("-a, --author <string>", "author"))
   .addOption(new Option("-d, --discussion <string>", "forum link"))
   .addOption(new Option("-s, --snapshot <string>", "snapshot link"))
-  .allowExcessArguments(false);
-
-program.parse();
+  .allowExcessArguments(false)
+  .parse(process.argv);
 
 const options = program.opts();
 
-const baseName = generateName(options);
+// workaround as there's validate is not currently supported on checkbox
+// https://github.com/SBoudrias/Inquirer.js/issues/1257
+while (!options.chains?.length === true) {
```

Comment:
Not very familiar with the packages you're using but is it possible to enter into an infinite loop here?

Reply:
it's in fact an intended infinite loop here, it will reask till you selected at least one chain :sweat_smile:

Reply:
Ah I see, makes sense!